homework/src/test/java/task05/data/TriangleDataProviders.java

```diff
@@ -0,0 +1,92 @@
+package task05.data;
+
+import org.testng.annotations.DataProvider;
+
+import java.math.BigDecimal;
+
+/**
+ * @author Oleg Baslak
+ * @version 1.0
+ * @since 03.11.2016
+ */
+public class TriangleDataProviders {
+
+    @DataProvider(name = "invalidTriangles")
+    public static Object[][] getInvalidTriangleData() {
+        return new Object[][]{
+                {BigDecimal.valueOf(0), BigDecimal.valueOf(1), BigDecimal.valueOf(1)},
+                {BigDecimal.valueOf(1), BigDecimal.valueOf(0), BigDecimal.valueOf(1)},
+                {BigDecimal.valueOf(1), BigDecimal.valueOf(1), BigDecimal.valueOf(0)},
+
+                {BigDecimal.valueOf(-1), BigDecimal.valueOf(1), BigDecimal.valueOf(1)},
+                {BigDecimal.valueOf(1), BigDecimal.valueOf(-1), BigDecimal.valueOf(1)},
+                {BigDecimal.valueOf(1), BigDecimal.valueOf(1), BigDecimal.valueOf(-1)},
+
+                {null, BigDecimal.valueOf(1), BigDecimal.valueOf(1)},
+                {BigDecimal.valueOf(1), null, BigDecimal.valueOf(1)},
+                {BigDecimal.valueOf(1), BigDecimal.valueOf(1), null}
+        };
+    }
+
+    @DataProvider(name = "validCommonTriangle")
+    public static Object[][] getValidCommonTriangleData() {
+        BigDecimal maxDouble = BigDecimal.valueOf(Double.MAX_VALUE);
```

Comment:
Just realized that you use BigDecimal as Triangle sides so Double.MAX_VALUE (and the similar) aren't in different equivalence classes so actually you don't need to test it